src/framework/MOM_restart.F90

```diff
@@ -1556,4 +1607,69 @@ subroutine restart_error(CS)
   endif
 end subroutine restart_error
 
+subroutine get_MOM_compute_domain(G,isL,ieL,jsL,jeL)
+! use mpp_domains_mod, only:  mpp_get_domain_shift
+  type(ocean_grid_type), intent(in)  :: G         !< The ocean's grid structure
+  integer , intent(out):: isL,ieL,jsL,jeL
+!  integer , intent(in) :: sizes(:),pos
+!  integer :: is0,js0
+!  integer ::  iadd,jadd,ishift, jshift, pos,sizes(7)
+
+  !Simplistic way
+  isL=G%isc-G%isd+1
+  ieL=G%iec-G%isd+1
+  jsL=G%jsc-G%jsd+1
+  jeL=G%jec-G%jsd+1
+  !Zhi's way
```

Comment:
I think that for the purpose of checksums the simplistic way is the only way. If @Hallberg-NOAA concurs with that statement, we should delete the commented out simplistic way.

The only time I can see the simplistic approach failing is for when symmetric-memory staggered data is written and the data on opposite sides of the domain are different (e.g. regional models). Do the FMS version of this handle that circumstance?

Reply:
I concur that this simplistic approach is the right one for determining the extent of the checksums. However, I am concerned that we are using the same array-extent variables for the checksums and for the fields to read, and I worry that in doing so we will now be reading in the wrong extents in some cases (e.g., reading in a symmetric memory restart file for a non-symmetric memory case, or vice versa).  This worked previously, but a visual inspection of the code suggests that this might have been broken.  Using different variables (e.g., isL and is_chk) for the extents for reading and checksums would seem to me to be more robust, precisely because they are _supposed_ to operate on different array extents with symmetric memory.